tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,192 @@
+    transformed_features.pop(transformed_name(LABEL_KEY), None)
+
+    return model(transformed_features)
+
+  return serve_tf_examples_fn
+
+def _input_fn(file_pattern: List[Text],
+              tf_transform_output: tft.TFTransformOutput,
+              batch_size: int = 200) -> tf.data.Dataset:
+  """Generates features and label for tuning/training.
+
+  Args:
+    file_pattern: List of paths or patterns of input tfrecord files.
+    tf_transform_output: A TFTransformOutput.
+    batch_size: representing the number of consecutive elements of returned
+      dataset to combine in a single batch
+
+  Returns:
+    A dataset that contains (features, indices) tuple where features is a
+      dictionary of Tensors, and indices is a single Tensor of label indices.
+  """
+  transformed_feature_spec = (
+      tf_transform_output.transformed_feature_spec().copy())
+
+  dataset = tf.data.experimental.make_batched_features_dataset(
+      file_pattern=file_pattern,
+      batch_size=batch_size,
+      features=transformed_feature_spec,
+      reader=_gzip_reader_fn,
+      label_key=transformed_name(LABEL_KEY))
+
+  return dataset
+
+def _build_keras_model() -> tf.keras.Model:
+  """Creates a MobileNet model pretrained on ImageNet for classifying
+  CIFAR10 images.
+
+  Returns:
+    A Keras Model.
+  """
+  tf.keras.backend.set_image_data_format('channels_last')
+  base_model = tf.keras.applications.MobileNet(
+      input_shape=(224, 224, 3), include_top=False, weights='imagenet',
+      pooling='avg')
+
+  # Freeze all layers in the base model except last conv block
+  for layer in base_model.layers:
+    if '13' not in layer.name:
+      layer.trainable = False
+
+  model = tf.keras.Sequential([
+      tf.keras.layers.InputLayer(
+          input_shape=(224, 224, 3), name=transformed_name(IMAGE_KEY)),
+      base_model,
+      tf.keras.layers.Dropout(0.3),
+      tf.keras.layers.Dense(10, activation='softmax')
+  ])
+
+  model.compile(
+      loss='sparse_categorical_crossentropy',
+      optimizer=tf.keras.optimizers.RMSprop(lr=0.0015),
+      metrics=['sparse_categorical_accuracy'])
+  model.summary(print_fn=absl.logging.info)
+  return model
+
+# TFX Transform will call this function.
+def preprocessing_fn(inputs):
+  """tf.transform's callback function for preprocessing inputs.
+
+  Args:
+    inputs: map from feature keys to raw not-yet-transformed features.
+
+  Returns:
+    Map from string feature key to transformed feature operations.
+  """
+  outputs = {}
+
+  image_features = tf.map_fn(lambda x: tf.io.decode_png(x[0], channels=3), inputs[IMAGE_KEY], dtype=tf.uint8)
+  image_features = tf.cast(image_features, tf.float32)
+
+  # The MobileNet we use was trained on ImageNet, which has image size 224 x 224.
+  # We resize CIFAR10 images to match that size
+  image_features = tf.image.resize(image_features, [224, 224])
+
+  image_features = tf.ensure_shape(image_features, (None, 224, 224, 3))
```

Comment:
Here it's just a check, because I did not use tft functions. It will be necessary if we want to use tf.map_fn and tft functions together, since tft functions require "all the dimensions except batch dimension should be known", and the tf.map_fn will make some other dimension to be None.